tensorflow/compiler/jit/build_xla_ops_pass.cc

```diff
@@ -135,9 +135,9 @@ void MergeOutgoingDataEdges(const Scope& s, Node* old_node, Node* new_node,
         new_output = check_numerics_op;
       }
 
-      ops::Merge merge_op(s.WithOpName("merge_oidx_", oidx),
-                          {Output(old_node, oidx), new_output});
-      merged_output = merged_outputs[oidx] = merge_op.output;
+      ops::_XlaMerge xla_merge_op(s.WithOpName("merge_oidx_", oidx),
```

Comment:
Does this work for `DT_RESOURCE` outputs?

Reply:
It should. haven't seen any issue locally.

Reply:
Looks like both strings and resources are handled specially and automatically placed on the host: https://github.com/tensorflow/tensorflow/blob/cca6113667e10b1f3405dbe3b1bd3d983f9cf420/tensorflow/core/framework/memory_types.cc#L146